tools/RAiDER/dem.py

```diff
@@ -243,12 +238,50 @@ def isInside(extent1, extent2):
     return False
 
 
-def getDEM(extent, inRaster = '/vsicurl/{}'.format(_world_dem)):
-    memRaster = '/vsimem/warpedDEM'
-    gdal_extent = [extent[2], extent[0], extent[3], extent[1]]
-    gdal.BuildVRT(memRaster, inRaster, outputBounds=gdal_extent)
-    out = gdal_open(memRaster)
-    out = out[::-1] # GDAL returns rasters upside-down
-    return out
+def getDEM(extent, out_dir):
+    r = requests.get(_DEM.format(*extent), allow_redirects=True)
+    dem_raster = get_filename_from_cd(r.headers.get('content-disposition'))
+    filename = os.path.join(out_dir, dem_raster)
+    open(filename, 'wb').write(r.content)
+    return filename
 
+    
+def get_filename_from_cd(cd):
+    """
+    Get filename from content-disposition
+    """
+    if not cd:
+        return None
+    fname = re.findall('filename=(.+)', cd)
+    if len(fname) == 0:
+        return None
+    return fname[0]
+
+
+def readRaster(filename, band_num = None):
+    '''
+    Read a GDAL VRT file and return its attributes
+    '''
+    try:
+        ds = gdal.Open(filename, gdal.GA_ReadOnly)
+        if ds is None:
+            raise RuntimeError('readRaster: cannot find file {}'.format(filename))
+    except Exception as e:
+        ds = None
+        raise RuntimeError('readRaster: cannot open file {}. Reason: {}'.format(filename, e))
+
+    xSize = ds.RasterXSize
+    ySize = ds.RasterYSize
+    geoProj = ds.GetProjection()
+    trans = ds.GetGeoTransform()
+    Nbands = ds.RasterCount
+    if band_num is None:
+        band_num = 1
+        print('Using band one for dataType')
+
+    dType = ds.GetRasterBand(band_num).DataType
+    noDataVal = ds.GetRasterBand(band_num).GetNoDataValue()
```

Comment:
This is ok when there is a no-data value associated with the file. 
Should have a catch for the case when that is not the case.

e.g. user provides a DEM as an envi file without that field.